src/Hint/Lambda.hs

```diff
@@ -149,11 +151,13 @@ lambdaDecl
 
           (finalpats, body) = fromLambda . lambda pats $ origBody
           (pats2, bod2) = etaReduce pats origBody
-          template fps = unsafePrettyPrint $ reform (zipWith munge ['a'..'z'] fps) varBody
-          subts fps b = ("body", toSS b) : zipWith (\x y -> ([x],y)) ['a'..'z'] (map toSS fps)
+          template fps =
```

Comment:
Why make template a function and then only apply it at finalpats? If it's just going to be finalpats, why not `(origPats, substsVars) = mkOrigPats finalpats` at the top level and avoid the nested lets?